ballerina/tests/test.bal

```diff
@@ -0,0 +1,312 @@
+    BatchResponsePublicDefaultAssociation|error response = hubspotAssociations->/associations/["dea"]/["com"]/batch/associate/default.post(
+        payload = {
+            inputs: [
+                {
+                    'from: {id: mockFromObjectId},
+                    to: {id: mockToObjectId}
+                }
+            ]
+        }
+    );
+    test:assertTrue(response is error, msg = "Expected an error response, but got a successful response.");
+}
+
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+isolated function testCreateCustomAssociationByInvalidObjectType() returns error? {
+    BatchResponseLabelsBetweenObjectPair|BatchResponseLabelsBetweenObjectPairWithErrors|error response = hubspotAssociations->/associations/["dea"]/["com"]/batch/create.post(
+        payload = {
+            inputs: [
+                {
+                    types: [
+                        {
+                            associationCategory: "USER_DEFINED",
+                            associationTypeId: 9
+                        }
+                    ],
+                    'from: {id: mockFromObjectId},
+                    to: {id: mockToObjectId}
+                }
+            ]
+        }
+    );
+    test:assertTrue(response is error, msg = "Expected an error response, but got a successful response.");
+}
+
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+isolated function testDeleteSpecificLablesByInvalidObjectType() returns error? {
+    http:Response|error response = hubspotAssociations->/associations/["dea"]/["com"]/batch/labels/archive.post(
+        payload = {
+            inputs: [
+                {
+                    types: [
+                        {
+                            associationCategory: "HUBSPOT_DEFINED",
+                            associationTypeId: 9
+                        }
+                    ],
+                    'from: {id: mockFromObjectId},
+                    to: {id: mockToObjectId}
+                }
+            ]
+        }
+    );
+    if response is http:Response {
+        test:assertEquals(response.statusCode, 400,
+                msg = string `Expected status code 400 but got ${response.statusCode}`);
+    }
+}
+
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+isolated function testDeleteAllAssociationsByInvalidObjectType() returns error? {
+    http:Response|error response = hubspotAssociations->/objects/["com"]/["38056537829"]/associations/["dea"]/["41479955131"].delete();
+    if response is http:Response {
+        test:assertEquals(response.statusCode, 400,
+                msg = string `Expected status code 400 but got ${response.statusCode}`);
+    }
+}
+
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+isolated function testRemoveAssociationBetweenObjectByInvalidObjectType() returns error? {
+    http:Response|error response = hubspotAssociations->/associations/["dea"]/["com"]/batch/archive.post(
```

Comment:
```suggestion
    http:Response response = check hubspotAssociations->/associations/["dea"]/["com"]/batch/archive.post(
```

You can use the `check` keyword here. That way you don't have to explicitly check whether the response is `http:Response` in the lines below